tasks.php

```diff
@@ -1692,22 +1692,31 @@ function RelatedPostings($tid)
     echo "<table class='themed theme_striped'>\n";
     $related_postings = decode_array($related_postings);
     asort($related_postings);
-    foreach ($related_postings as $val) {
-        $row = get_topic_details($val);
-        $forum_url = get_url_to_view_forum($row["forum_id"]);
-        $topic_url = get_url_to_view_topic($row["topic_id"]);
+    foreach ($related_postings as $topic_id) {
 
         echo "<tr><td>";
-        echo "<a href='$forum_url'>" . $row['forum_name'] . "</a>";
-        echo "&nbsp;&raquo;&nbsp;";
-        echo "<a href='$topic_url'>" . $row['title'] . "</a>";
-        echo " (" . sprintf(_('Posted by: %1$s - %2$d replies'), $row['creator_username'], $row['num_replies']) . ")\n";
+
+        if (does_topic_exist($topic_id)) {
+            $row = get_topic_details($topic_id);
+            $forum_url = get_url_to_view_forum($row["forum_id"]);
+            $topic_url = get_url_to_view_topic($row["topic_id"]);
+            echo "<a href='$forum_url'>" . $row['forum_name'] . "</a>";
+            echo "&nbsp;&raquo;&nbsp;";
+            echo "<a href='$topic_url'>" . $row['title'] . "</a>";
```

Comment:
Do we want to html encode the user provided content (post title) to avoid any XSS/rendering issues?

Reply:
Surprisingly no because phpBB escapes HTML entities upon save, so they are escaped in the database:
```
mysql> select topic_title from phpbb_topics where topic_id=53;
+----------------------------------------------------+
| topic_title                                        |
+----------------------------------------------------+
| &lt;i&gt;i&lt;/i&gt; &amp;amp; ampersand &lt;b&gt; |
+----------------------------------------------------+
1 row in set (0.00 sec)
```
for [topic 53](https://www.pgdp.org/phpBB3/viewtopic.php?f=1&t=53&p=113#p113). Good catch though because my first thought was yes we should!